examples/filesaver/main.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	lksdk "github.com/livekit/server-sdk-go"
+	"github.com/livekit/server-sdk-go/pkg/samplebuilder"
+	"github.com/pion/rtp/codecs"
+	"github.com/pion/webrtc/v3"
+	"github.com/pion/webrtc/v3/pkg/media"
+	"github.com/pion/webrtc/v3/pkg/media/h264writer"
+	"github.com/pion/webrtc/v3/pkg/media/ivfwriter"
+	"github.com/pion/webrtc/v3/pkg/media/oggwriter"
+)
+
+var (
+	host, apiKey, apiSecret, roomName, identity string
+)
+
+func init() {
+	flag.StringVar(&host, "host", "", "livekit server host")
+	flag.StringVar(&apiKey, "api-key", "", "livekit api key")
+	flag.StringVar(&apiSecret, "api-secret", "", "livekit api secret")
+	flag.StringVar(&roomName, "room-name", "", "room name")
+	flag.StringVar(&identity, "identity", "", "participant identity")
+}
+
+func main() {
+	flag.Parse()
+	if host == "" || apiKey == "" || apiSecret == "" || roomName == "" || identity == "" {
+		fmt.Println("invalid arguments.")
+		return
+	}
+	room, err := lksdk.ConnectToRoom(host, lksdk.ConnectInfo{
+		APIKey:              apiKey,
+		APISecret:           apiSecret,
+		RoomName:            roomName,
+		ParticipantIdentity: identity,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	room.Callback.OnTrackSubscribed = trackSubscribed
+	// room.Callback.OnTrackUnsubscribed = nil
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT)
+
+	<-sigChan
+	room.Disconnect()
+}
+
+func trackSubscribed(track *webrtc.TrackRemote, publication *lksdk.RemoteTrackPublication, rp *lksdk.RemoteParticipant) {
+	fileName := fmt.Sprintf("%s-%s", rp.Identity(), track.ID())
+	fmt.Println("write track to file ", fileName)
+	NewTrackWriter(track, rp.WritePLI, fileName)
+}
+
+const (
+	maxLate = 200
```

Comment:
can we increase this to 2s or something very high? since for recording it doesn't really care how behind it is

Reply:
this is a packet sequence latency, 200 packets means 4s for audio. 2s for medium quality video. ideally we can optimize this latency to seconds later

Reply:
use two latency for audio/video now